core/src/main/java/org/apache/iceberg/SystemConfigs.java

```diff
@@ -42,6 +42,13 @@ private SystemConfigs() {}
           Math.max(2, Runtime.getRuntime().availableProcessors()),
           Integer::parseUnsignedInt);
 
+  public static final ConfigEntry<Integer> DELETE_WORKER_THREAD_POOL_SIZE =
+      new ConfigEntry<>(
+          "iceberg.worker.delete-num-threads",
+          "ICEBERG_WORKER_DELETE_NUM_THREADS",
+          4 * Runtime.getRuntime().availableProcessors(),
```

Comment:
This value may sound ridiculous but here is my thought process: there is one such thread pool per JVM, each core in an executor can get a data task that may need to load 1 to many delete files, these tasks are I/O intensive. This value essentially means we can try to load 4 delete files concurrently per each data task.

Reply:
Yeah, it is configurable, we just need to make sure the default value is reasonable.